src/sentry/remote_config/endpoints.py

```diff
@@ -0,0 +1,73 @@
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint, ProjectEventPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.models.project import Project
+from sentry.models.projectkey import ProjectKey
+from sentry.remote_config.storage import make_storage
+
+
+class ConfigurationValidator(Serializer):
+    id = serializers.UUIDField(read_only=True)
+    sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    traces_sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    user_config = serializers.JSONField(required=True, allow_null=True)
+
+
+class ConfigurationContainerValidator(Serializer):
+    data = ConfigurationValidator()  # type: ignore[assignment]
+
+
+@region_silo_endpoint
+class ProjectConfigurationEndpoint(ProjectEndpoint):
+    owner = ApiOwner.REMOTE_CONFIG
+    permission_classes = (ProjectEventPermission,)
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def convert_args(self, request: Request, key_id: str, *args, **kwargs):
+        args, kwargs = super().convert_args(request, *args, **kwargs)
+        project = kwargs["project"]
+
+        try:
+            key = ProjectKey.objects.for_request(request).get(project=project, public_key=key_id)
```

Comment:
If key_id = project's public key, where's the name `key_id` come from?

Reply:
Conventional behavior taken from other endpoints which share the route.  Public key could be more contextual but I'm not too worried about it.  Open to change if you feel its important.